modules/tensor_mechanics/src/materials/cohesive_zone_model/SlipWeakeningFriction2d.C

```diff
@@ -0,0 +1,163 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+/*
+Material Description of Slip Weakening Friction Law 2D
+*/
+
+#include "SlipWeakeningFriction2d.h"
+#include "InterfaceKernel.h"
+
+registerMooseObject("TensorMechanicsApp", SlipWeakeningFriction2d);
+
+InputParameters
+SlipWeakeningFriction2d::validParams()
+{
+  InputParameters params = CZMComputeLocalTractionTotalBase::validParams();
+  params.addClassDescription("Linear Slip Weakening Traction Separation Law.");
+  params.addParam<Real>("T2_o", 1.0, "Background normal traction");
```

Comment:
Should these have default values or should the user be required to provide them.  I think they should be required.

Reply:
It should be required yes